Packs/CyrenInboxSecurity/Integrations/CyrenInboxSecurity/CyrenInboxSecurity.py

```diff
@@ -0,0 +1,504 @@
+         a key and a timestamp in seconds as a value
+        first_fetch_time (datetime): datetime object
+         on when to start fetching incidents
+        max_fetch (int): Maximum numbers of incidents per fetch
+    Returns:
+        incidents: Incidents that will be created in Cortex XSOAR
+    """
+    # last run is a dict with last_fetch as a key and
+    #  a timestamp in seconds as a value
+    last_fetch = last_run.get('last_fetch')
+
+    # Handle first fetch time
+    if last_fetch is None:
+        # if missing, use what provided via first_fetch_time
+        date_from = int(date_to_timestamp(first_fetch_time) / 1000)
+    else:
+        # otherwise use the stored last fetch
+        date_from = last_fetch
+
+    last_fetched_time = date_from
+
+    incidents = []
+    token = client.get_token(client_id, client_secret).\
+        get('data').get('access_token')
+    cyren_incidents = client.get_incidents(token, date_from, max_fetch)
+    for ci in cyren_incidents.get('data', []):
+        incident_name = 'Cyren Inbox Security - {} ({})'.format(
+            ci.get('threat_type', 'phishing'), ci.get('reported_by', 'System')
+        )
+        incident_timestamp = ci.get('created_at')
+        incident_str_time = timestamp_to_datestring(incident_timestamp * 1000)
+        incident = {
+            'name': incident_name,
+            'occurred': incident_str_time,
+            'rawJSON': json.dumps(ci),
+            'severity': convert_to_demisto_severity(ci.get('confidence', 1)),
+        }
+
+        incidents.append(incident)
+
+        # Update last_fetch if the incident is newer than last_fetch
+        if incident_timestamp > last_fetched_time:
+            last_fetched_time = incident_timestamp + 1
+
+    demisto.setLastRun({
+        'last_fetch': last_fetched_time
+    })
+
+    return incidents
+
+
+def resolve_and_remediate_command(client, args, client_id, client_secret):
+    """
+    Resolve a case and remediate it
+    """
+    case_id = args.get('case_id')
+    resolution = args.get('resolution', 'phishing')
+    resolution_reason = args.get('resolution_reason', '')
+    resolution_reason_text = args.get('resolution_reason_text', '')
+    actions = args.get('actions', [])
+    if actions:
+        actions = actions.split(',')
+
+    if client_id == client_secret == "sample":
+        resp = {"data": {"status": "ok"}}
+    else:
+        token = client.get_token(client_id, client_secret).\
+            get('data').get('access_token')
+        resp = client.resolve_and_remediate(
+            token,
+            case_id,
+            resolution=resolution,
+            resolution_reason=resolution_reason,
+            resolution_reason_text=resolution_reason_text,
+            actions=actions
+        )
+
+    readable_output = (
+        tableToMarkdown("cyren-resolve-and-remediate results", resp["data"])
+        + '\n*** end of results ***'
+    )
+
+    return CommandResults(
```

Comment:
As we talked about in the demo, please handle the error messages in each command in order to bring the user as many details as possible in a readable way, in case of a returned error.